### ISSUE TITLE
Design Hyperparameter Search Strategy for MLP Probes

### ISSUE THREAD

**Issue by druhinb** [druhinb/LatentInvestigation#12]
Define a systematic approach for tuning our MLP probe architectures (depth, width, activation) and training hyperparameters (learning rate, weight decay) for each task.

**Subtasks**

- Select a hyperparameter optimization strategy (Optuna is a good start)
- Define search spaces for MLP parameters.
- Expose as a training wrapper we can use during the MLP-related experiments
